src/xia2/Modules/MultiCrystal/data_manager.py

```diff
@@ -22,7 +22,12 @@
 
 
 class DataManager:
-    def __init__(self, experiments: ExperimentList, reflections: flex.reflection_table):
+    def __init__(
+        self,
+        experiments: ExperimentList,
+        reflections: flex.reflection_table,
+        batch_offset_list: list[int] = [],
```

Comment:
It is dangerous to use a mutable default argument in a function in python, instead it should be default initialised to None like `batch_offset_list: list[int] | None = None`